OpenNefia.Content/UI/Hud/HudLayer.cs

```diff
@@ -1,56 +1,371 @@
+        {
+            return stat.Buffed.CompareTo(stat.Base) switch
+            {
+                < 0 => Color.Red,
+                > 0 => Color.Green,
+                _ => Color.Black
+            };
+        }
+
+        private string GetIconRegion(HudSkillIconType type)
+        {
+            return $"{(int)type}";
+        }
+
+        public void ToggleBacklog(bool visible)
+        {
+            ShowingBacklog = visible;
         }
 
         public override void SetSize(int width, int height)
         {
             base.SetSize(width, height);
-            MessageWindow.SetSize(_graphics.WindowSize.X - 100, 150);
+            Minimap.SetSize(MinimapWidth - 4, MinimapHeight - 4);
+            MessageWindow.SetSize(Width - MinimapWidth - 80, MessageWindow.Height);
             FpsCounter.SetSize(400, 500);
+            MessageBoxBacking.SetSize(Width + MinimapWidth, 72);
+            BacklogBacking.SetSize(width + MinimapWidth, 600);
+            HudBar.SetSize(Width + MinimapWidth, UiHudBar.HudBarHeight);
+            DateText.SetPreferredSize();
+            HealthText.SetPreferredSize();
+            ManaText.SetPreferredSize();
         }
 
         public override void SetPosition(int x, int y)
         {
             base.SetPosition(x, y);
+            Minimap.SetPosition(2, Height - MinimapHeight + 2);
             MessageWindow.SetPosition(X + 50, Y + Height - MessageWindow.Height - 10);
             FpsCounter.SetPosition(Width - FpsCounter.Text.Width - 5, 5);
+            MessageBoxBacking.SetPosition(MinimapWidth, Height - MinimapHeight);
+            BacklogBacking.SetPosition(MinimapWidth + 10, Height - 470);
+            BacklogContainer.SetPosition(MinimapWidth + 15, BacklogBacking.Y + 12);
+            HudBar.SetPosition(MinimapWidth, Height - 18);
+            MessageBoxContainer.SetPosition(BacklogContainer.X, Height - MinimapHeight + 4);
+            DateText.SetPosition(120, 17);
+            ClockHand.SetPosition(62, 48);
+            HealthText.SetPosition(280, Height - MinimapHeight - 19);
+            ManaText.SetPosition(420, Height - MinimapHeight - 19);
+            ExpText.SetPosition(35, Height - MinimapHeight - 13);
+            GoldText.SetPosition(590, Height - MinimapHeight - 13);
+            PlatText.SetPosition(700, Height - MinimapHeight - 13);
+            HudSkillContainer.SetPosition(305, Height - 14);
+            HudSkillContainer.Relayout();
+            MapNameText.SetPosition(160, Height - 14);
         }
 
         public override void Update(float dt)
         {
             MessageWindow.Update(dt);
             FpsCounter.Update(dt);
+            MessageBoxContainer.Update(dt);
+            BacklogContainer.Update(dt);
+            DateText.Update(dt);
+            ClockHand.Update(dt);
         }
 
         public override void Draw()
         {
+            GraphicsEx.SetColor(Color.White);
+            MiniMapAsset.Draw(0, Height - MinimapHeight);
+            Minimap.Draw();
```

Comment:
I think you can draw the `MiniMapAsset` in thie `MiniMap` itself.

Reply:
changed in 9eb449f
kept the Minimap itself seperate from the widget so you could still render it in different resolutions if wanted